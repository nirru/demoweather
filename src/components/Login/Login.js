import React from 'react';
import './login.css';
import {getCurrentUserInfo, setCurrentUser} from '../../actions';
import {connect} from 'react-redux';

class Login extends React.Component {

  constructor(props){
    super(props);
    this.state= {
      email:'',
      password:'',
    };
  }

  componentDidMount() {
    localStorage.removeItem('USER');
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  callback = e => {
    e.preventDefault();
    console.log('CLICK')
    const {dispatch} = this.props;
    dispatch(getCurrentUserInfo(this.state));
    // dispatch(setCurrentUser(this.state));
  };
  render() {
    return (
      <div className="row">
        <div className="col-sm-9 col-md-7 col-lg-5 mx-auto">
          <div className="card card-signin my-5">
            <div className="card-body">
              <h5 className="card-title text-center">Sign In</h5>
              <form className="form-signin" onSubmit={this.callback}>
                <div className="form-label-group">
                  <input type="email" id="inputEmail" className="form-control" placeholder="Email address" required
                    autoFocus onChange={this.handleChange('email')} value={this.state.email}/>
                  <label htmlFor="inputEmail">Email address</label>
                </div>

                <div className="form-label-group">
                  <input type="password" id="inputPassword" className="form-control" placeholder="Password" required onChange={this.handleChange('password')}/>
                  <label htmlFor="inputPassword">Password</label>
                </div>

                <div className="custom-control custom-checkbox mb-3">
                  <input type="checkbox" className="custom-control-input" id="customCheck1"/>
                  <label className="custom-control-label" htmlFor="customCheck1">Remember password</label>
                </div>
                <button className="btn btn-lg btn-primary btn-block text-uppercase" type="submit">Sign in</button>
                <hr className="my-4"/>
                {/*<button className="btn btn-lg btn-google btn-block text-uppercase" type="submit"><i*/}
                {/*  className="fab fa-google mr-2"></i> Sign in with Google*/}
                {/*</button>*/}
                {/*<button className="btn btn-lg btn-facebook btn-block text-uppercase" type="submit"><i*/}
                {/*  className="fab fa-facebook-f mr-2"></i> Sign in with Facebook*/}
                {/*</button>*/}
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}


export default connect(undefined,undefined)(Login);