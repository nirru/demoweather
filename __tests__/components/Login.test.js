import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { spy } from 'sinon';
// Component to be tested
import Login from '../../lib/components/Login/Login';

describe('Login Component', () => {

  // make our assertion and what we expect to happen
  // eslint-disable-next-line no-undef
  it('should render without throwing an error', () => {
    // eslint-disable-next-line no-undef
    expect(shallow(<Login />).find('form.form-signin').exists()).toBe(true);
  });

});


describe('Renders email input', () => {
  it ('renders a email input', () => {
    expect(shallow(<Login />).find('#inputEmail').length).toEqual(1);
  });
});

describe('Renders password input', () => {
  it ('renders a Password input', () => {
    expect(shallow(<Login />).find('#inputPassword').length).toEqual(1);
  });
});

describe ('Email input', () => {

  it('should respond to change event and change the state of the Login Component', () => {

    const wrapper = shallow(<Login />);
    wrapper.find('#inputEmail').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}});

    expect(wrapper.state('email')).toEqual('blah@gmail.com');
  });
});

describe('Password input', () => {

  it('should respond to change event and change the state of the Login Component', () => {

    const wrapper = shallow(<Login />);
    wrapper.find('#inputPassword').simulate('change', {target: {name: 'password', value: 'cats'}});

    expect(wrapper.state('password')).toEqual('cats');
  });
});

describe('Form', () => {
  it('submit event when click submit', () => {
    const wrapper = mount(<Login />);
    wrapper.find('form').simulate('submit');
  });
});

