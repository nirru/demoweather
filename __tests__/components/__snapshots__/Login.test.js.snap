// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render() renders the component 1`] = `
<div
  className="row"
>
  <div
    className="col-sm-9 col-md-7 col-lg-5 mx-auto"
  >
    <div
      className="card card-signin my-5"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title text-center"
        >
          Sign In
        </h5>
        <form
          className="form-signin"
        >
          <div
            className="form-label-group"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="inputEmail"
              placeholder="Email address"
              required={true}
              type="email"
            />
            <label
              htmlFor="inputEmail"
            >
              Email address
            </label>
          </div>
          <div
            className="form-label-group"
          >
            <input
              className="form-control"
              id="inputPassword"
              placeholder="Password"
              required={true}
              type="password"
            />
            <label
              htmlFor="inputPassword"
            >
              Password
            </label>
          </div>
          <div
            className="custom-control custom-checkbox mb-3"
          >
            <input
              className="custom-control-input"
              id="customCheck1"
              type="checkbox"
            />
            <label
              className="custom-control-label"
              htmlFor="customCheck1"
            >
              Remember password
            </label>
          </div>
          <button
            className="btn btn-lg btn-primary btn-block text-uppercase"
            type="submit"
          >
            Sign in
          </button>
          <hr
            className="my-4"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
